src/lib.rs

```diff
@@ -78,117 +79,139 @@ where
-    /// 
-    /// The output value is the value in degrees Celcius multiplied by 100.
-    pub fn read_default_conversion(&mut self) -> Result<u32, E> {
+    ///
+    /// The output value is the value in Ohms multiplied by 100.
+    pub fn read_ohms(&mut self) -> Result<u32, Error<E>> {
         let raw = self.read_raw()?;
         let ohms = ((raw >> 1) as u32 * self.calibration) >> 15;
-        let temp = temp_conversion::lookup_temperature(ohms as u16);
+
+        Ok(ohms)
+    }
+
+    /// Read the raw resistance value and then perform conversion to degrees Celsius.
+    ///
+    /// # Remarks
+    ///
+    /// The output value is the value in degrees Celsius multiplied by 100.
+    pub fn read_default_conversion(&mut self) -> Result<i32, Error<E>> {
+        let ohms = self.read_ohms()?;
+        let temp = temp_conversion::LOOKUP_VEC_PT100.lookup_temperature(ohms as i32);
 
         Ok(temp)
     }
 
     /// Read the raw RTD value.
-    /// 
+    ///
     /// # Remarks
-    /// 
+    ///
     /// The raw value is the value of the combined MSB and LSB registers.
     /// The first 15 bits specify the ohmic value in relation to the reference
     /// resistor (i.e. 2^15 - 1 would be the exact same resistance as the reference
     /// resistor). See manual for further information.
-    /// The last bit specifies if the conversion was successful. 
-    pub fn read_raw(&mut self) -> Result<u16, E> {
-        let msb : u16 = self.read(Register::RTD_MSB)? as u16;
-        let lsb : u16 = self.read(Register::RTD_LSB)? as u16;
-        
+    /// The last bit specifies if the conversion was successful.
+    pub fn read_raw(&mut self) -> Result<u16, Error<E>> {
+        let msb: u16 = self.read(Register::RTD_MSB)? as u16;
+        let lsb: u16 = self.read(Register::RTD_LSB)? as u16;
+
         Ok((msb << 8) | lsb)
     }
 
     /// Determine if a new conversion is available
-    /// 
+    ///
     /// # Remarks
-    /// 
-    /// When the module is finished converting the temperature it sets the 
-    /// ready pin to low. It is automatically returned to high upon reading the 
+    ///
+    /// When the module is finished converting the temperature it sets the
+    /// ready pin to low. It is automatically returned to high upon reading the
     /// RTD registers.
-    pub fn is_ready(&self) -> Result<bool, E> {
-        Ok(self.rdy.is_low())
+    pub fn is_ready(&self) -> Result<bool, RDY::Error> {
+        self.rdy.is_low()
     }
 
-    fn read(&mut self, reg: Register) -> Result<u8, E> {
+    fn read(&mut self, reg: Register) -> Result<u8, Error<E>> {
         let buffer: [u8; 2] = self.read_many(reg)?;
         Ok(buffer[1])
     }
 
-    fn read_many<B>(&mut self, reg: Register) -> Result<B, E> 
-    where B: Unsize<[u8]>,
+    fn read_many<B>(&mut self, reg: Register) -> Result<B, Error<E>>
+    where
+        B: Unsize<[u8]>,
     {
-        let mut buffer: B  = unsafe { mem::zeroed() };
+        let mut buffer: B = unsafe { mem::zeroed() };
```

Comment:
`fn read_many` is not pub and used only in a place to read `[u8; 2]`, what do you think of removing the generics and just read 2 bytes, this would allow removing the `unsafe` and also the `#![feature(unsize)]`, not requiring nightly anymore

Reply:
Done